vernac/declare.ml

```diff
@@ -2150,7 +2165,8 @@ let save_lemma_admitted_delayed ~pm ~proof =
 let save_lemma_proved_delayed ~pm ~proof ~idopt =
   (* vio2vo used to call this with invalid [pinfo], now it should work fine. *)
   let pinfo = process_idopt_for_save ~idopt proof.pinfo in
-  finalize_proof ~pm proof pinfo
+  let ending = CEphemeron.default pinfo.proof_ending Proof_ending.Regular in
```

Comment:
Same comment about the extra `ending` parameter where we are already passing `pinfo` to the functions.

Reply:
right, but it seems out of place since some choices that regard the ending need to be taken before that time. In particular I don't need the full ending info, just if is the equation/deriving case, which are ugly hacks, see where I use this extra ending.

Anyway, I can rectify that.

Reply:
done